src/java/net/jpountz/lz4/LZ4BlockInputStream.java

```diff
@@ -263,16 +261,25 @@ private void refill() throws IOException {
     o = 0;
   }
 
-  private void readFully(byte[] b, int len) throws IOException {
+  // Like readFully(), except it signals incomplete reads by returning
+  // false instead of throwing EOFException.
+  private boolean tryReadFully(byte[] b, int len) throws IOException {
```

Comment:
nit: drop `throws IOException` in the end.

Reply:
I considered this, but it will not work: the underlying `read()` call declares that it `throws IOException` so we need to keep this here.

Reply:
ah, ok.